lib/vrl/stdlib/src/encode_percent.rs

```diff
@@ -46,6 +64,20 @@ const WWW_FORM_URLENCODED: &AsciiSet =
 #[derive(Clone, Copy, Debug)]
 pub struct EncodePercent;
 
+fn ascii_sets() -> Vec<Value> {
```

Comment:
It looks like you might be able to return a static slice here and avoid the allocation, or is the problem that `value!` needs to allocate? Looks like the call sites for the value returned here then slice it anyway.

Reply:
It is because the `InvalidEnumVariant` error message that is created when you specify an invalid variant expects a Vec.

But on reflection we should return a static slice and then call `to_vec` for the unfortunate times when we need to error.

Reply:
Actually it is more an issue with Bytes rather than Value. I could possibly change it to be `fn ascii_sets() -> &'static [&'static Value]` but given that this is only a compile time thing I'm not convinced there is much value in optimising it here.